backend/volcon/auth/auth_bp.py

```diff
@@ -0,0 +1,57 @@
+from flask import jsonify, request, Blueprint, session
+from sqlalchemy.exc import SQLAlchemyError
+from werkzeug.security import generate_password_hash
+from volcon_db import db, Volunteer, Organization, User
+
+auth_bp = Blueprint('auth', __name__, url_prefix='/api/v1/auth')
+
+@auth_bp.route('/signup', methods=['POST'])
+def signup():
+    try:
+        data = request.get_json()
+        email = data['email']
+        password = data['password']
+        role = data['role']
+        if role == 'volunteer':
+            user = Volunteer(
+                email=email,
+                password=generate_password_hash(password, method='sha256')
+            )
+        elif role == 'organization':
+            user = Organization(
+                email=email,
+                password=generate_password_hash(password, method='sha256')
+            )
+        else:
+            return jsonify({'error': 'Invalid role specified.'}), 400
+        db.session.add(user)
+        db.session.commit()
+        return jsonify({'message': 'User registered successfully.'}), 201
+    except SQLAlchemyError as e:
+        error = str(e.__dict__.get('orig', e))
+        return jsonify({'error': error}), 500
+
+
+@auth_bp.route('/signin', methods=['POST'])
+"""A sign-in API route Complete with Session Management"""
+def login():
+    # retrieve credentials from request body
+    data = request.get_json()
+    username = data.get('username')
+    password = data.get('password')
+
+    # credentials verification against database
+    volunteer = Volunteer.query.filter_by(username=username, password=password).first()
+    organization = Organization.query.filter_by(username=username, password=password).first()
+
+    # check if user exists and assign session ID
+    if volunteer:
+        session['user_id'] = volunteer.volunteer_id
+        session['role'] = 'volunteer'
+        return jsonify({'message': 'Login successful'})
```

Comment:
What you'll return here is a json of the generated token & user information. install this library (`flask_jwt_extended`) to generate the token. follow this tutorial https://dev.to/nagatodev/how-to-add-login-authentication-to-a-flask-and-react-application-23i7